src/inputs/unifi_input.py

```diff
@@ -0,0 +1,173 @@
+            self.cameras = {}
+            for camera in data['cameras']:
+                channels = {channel['name']: {
+                    'id': channel['id'],
+                    'name': channel['name'],
+                    'enabled': channel['enabled'],
+                    'rtspAlias': channel['rtspAlias']
+                } for channel in camera['channels']}
+                self.cameras[camera['id']] = {
+                    'name': camera['name'].replace(" ", ""),
+                    'id': camera['id'],
+                    'mac': camera['mac'],
+                    'state': camera['state'],
+                    'lastMotion': camera['lastMotion'],
+                    'channels': channels
+                }
+        except Exception as e:
+            logging.error(f'[Unifi] Failed to bootstrap: {e}')
+            raise e
+
+    def get_rtsp_link(self, rtsp_alias: str) -> str:
+        """Generate an RTSP link for a camera."""
+        return self.host.replace('https', 'rtspx') + f':7441/{rtsp_alias}'
+
+    @staticmethod
+    def decode_packet(packet: bytes):
+        """Decode a received WebSocket packet."""
+        s = packet.decode('utf-8', errors='ignore')
+        json_objects = []
+        start_idx = 0
+        while True:
+            start = s.find('{', start_idx)
+            if start == -1:
+                break
+            brace_count = 1
+            end = start
+            while brace_count > 0 and end < len(s) - 1:
+                end += 1
+                if s[end] == '{':
+                    brace_count += 1
+                elif s[end] == '}':
+                    brace_count -= 1
+            if brace_count == 0:
+                json_str = s[start:end + 1]
+                try:
+                    json_obj = json.loads(json_str)
+                    json_objects.append(json_obj)
+                except json.JSONDecodeError:
+                    pass
+            start_idx = end + 1
+        return {'header': json_objects[0], 'payload': json_objects[1]}
+
+    def on_message(self, message) -> None:
+        """Handle messages received from WebSocket."""
+        decoded_message = self.decode_packet(message)
+        header = decoded_message.get('header')
+        payload = decoded_message.get('payload')
+        is_smart_detection = payload.get('isSmartDetected')
+        if payload and is_smart_detection:
+            camera_name = self.get_camera_name(header)
+            logging.debug(f'[Unifi] Smart motion start detected for name: {camera_name}')
+            super().start_capture_topic(camera_name)
+        elif payload and is_smart_detection == False:
+            camera_name = self.get_camera_name(header)
+            logging.debug(f'[Unifi] Smart motion stop detected for name: {camera_name}')
+            super().stop_capture_topic(camera_name)
+
+    def get_camera_name(self, header) -> None:
+        return f"{self.cameras[header.get('id')].get('name')}_{PF_UNIFI_IMAGE_QUALITY}".lower()
```

Comment:
There is already a configuration file, I think that it could be better to take it from there

Reply:
what config file you're talking about? cameras.json?

Reply:
sure, the cameras.json

I think for the first phase we could configure the same image quality for all the cameras